src/Compilers/CSharp/Portable/Errors/CSDiagnosticInfo.cs

```diff
@@ -34,7 +35,7 @@ internal CSDiagnosticInfo(ErrorCode code, object[] args, ImmutableArray<Symbol>
         {
             // Internal errors are abnormal and should not occur except where there are bugs in the compiler.
             Debug.Assert(code != ErrorCode.ERR_InternalError);
-            _additionalLocations = additionalLocations;
```

Comment:
Shouldn't we be changing the type of the `additionalLocations` parameter to be `IReadOnlyList<Location>` as well? Leaving it as `ImmutableArray<Location>` means callers could be unnecessarily allocating an `ImmutableArray<Location>`.

Reply:
In theory, maybe. In practice I did not observe this.